Replace IWDG_STM32_START_AT_BOOT by WDT_DISABLE_AT_BOOT

FRASTM opened the pull request:
With this patch, the flag specific flag IWDG_STM32_START_AT_BOOT
is replaced by the zephyr generic WDT_DISABLE_AT_BOOT.

Fix https://github.com/zephyrproject-rtos/zephyr/issues/21200

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
tested on nucleo_wb55rg 
- test suite kernel_critical  
- test suite wdt_basic_test

erwango commented:
@FRASTM, can you check this is in line with https://github.com/zephyrproject-rtos/zephyr/pull/22859/files ?

FRASTM committed:
drivers: watchdog: disable iwdg at boot on stm32

With this patch, the specific flag IWDG_STM32_START_AT_BOOT
is replaced by the zephyr generic WDT_DISABLE_AT_BOOT.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
tested with 
CONFIG_WDT_DISABLE_AT_BOOT=n
and 
CONFIG_WDT_DISABLE_AT_BOOT=y
in the nucleo_wb55rg_defconfig
